ethers/contract.nim

```diff
@@ -116,14 +117,46 @@ proc call(contract: Contract,
   let response = await contract.provider.call(transaction, overrides)
   return decodeResponse(ReturnType, returnMultiple, response)
 
+proc populateTransaction(
+  contract: Contract,
+  tx: Transaction
+): Future[Transaction] {.async.} =
+
+  if signer =? contract.signer:
+    try:
+      return await signer.populateTransaction(tx)
+    except EstimateGasError as e:
+      if nonce =? e.transaction.nonce:
+
+        if lastSeenNonce =? signer.lastSeenNonce and
+          lastSeenNonce > nonce:
+            discard await signer.cancelTransaction(e.transaction)
+            let revertReason = if not e.parent.isNil: e.parent.msg
+                               else: "unknown"
+            info "A cancellation transaction has been sent to prevent stuck " &
```

Comment:
Chronicles is not currently a dependency of nim-ethers (and I'd prefer it if we keep it that way :)

Reply:
After our discussion about removing chronicles, I've removed it from this PR, but it is still in #57.